src/actions/pagerduty/pagerduty.ts

```diff
@@ -0,0 +1,110 @@
+import * as Hub from "../../hub"
+import { PagerDutyClient } from './pagerduty_client'
+import { PagerDutyForm } from './pagerduty_form'
+import { PagerDutyCreateIncident } from "./pagerduty_create_incident"
+import * as moment from 'moment'
+import { ValidationError } from "../../hub"
+
+export class PagerDutyAction extends Hub.Action {
+  name = "pagerduty"
+  label = "PagerDuty"
+  iconName = "pagerduty/pd_icon.png"
+  description = "PagerDuty"
+  supportedActionTypes = [Hub.ActionType.Query, Hub.ActionType.Dashboard]
+  requiredFields = []
+  params = [{
+    name: "pagerduty_api_key",
+    label: "PagerDuty API Key",
+    required: true,
+    description: `A PagerDuty API key used to authenticate with the PagerDuty REST API. For how to generate a key, please follow instructions at https://support.pagerduty.com/docs/generating-api-keys`,
+    sensitive: true,
+  }]
+  usesStreaming = false
+
+  async execute(request: Hub.ActionRequest): Promise<Hub.ActionResponse> {
+    function validationError(e: Hub.ValidationError): Hub.ActionResponse {
+      return new Hub.ActionResponse({ message: 'Validation error', success: false, validationErrors: [e] })
+    }
+
+    const form = PagerDutyForm.create(request.formParams as unknown)
+    if (form == null)
+      return new Hub.ActionResponse({ message: 'Invalid form specification, please contact administrator', success: false, })
+
+    const apiKey = this.getApiKey(request)
+    const resultSet = this.parseResultSet(request.attachment)
+
+    if (typeof apiKey !== "string")
+      return validationError(apiKey)
```

Comment:
`apiKey` is of type `ValidationError | string`. `ValidationError` is an interface. This conditional would read better if something like this compiled:

```
if (apiKey instanceOf ValidationError) return validationError(apiKey)
```